<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QWidget" name="gridWidget" native="true">
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>2</number>
       </property>
       <property name="topMargin">
        <number>2</number>
       </property>
       <property name="rightMargin">
        <number>2</number>
       </property>
       <property name="bottomMargin">
        <number>2</number>
       </property>
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="minimumSize">
              <size>
               <width>370</width>
               <height>90</height>
              </size>
             </property>
             <property name="maximumSize">
              <size>
               <width>370</width>
               <height>90</height>
              </size>
             </property>
             <property name="autoFillBackground">
              <bool>false</bool>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="scaledContents">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTabWidget" name="tabWidget">
           <property name="minimumSize">
            <size>
             <width>250</width>
             <height>300</height>
            </size>
           </property>
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="tab">
            <attribute name="title">
             <string>引擎结构</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QTabWidget" name="tabWidget_2">
               <property name="tabPosition">
                <enum>QTabWidget::North</enum>
               </property>
               <property name="tabShape">
                <enum>QTabWidget::Rounded</enum>
               </property>
               <property name="currentIndex">
                <number>0</number>
               </property>
               <widget class="QWidget" name="tab_3">
                <attribute name="title">
                 <string>新建组件</string>
                </attribute>
                <layout class="QHBoxLayout" name="horizontalLayout_5">
                 <item>
                  <spacer name="horizontalSpacer_5">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeType">
                    <enum>QSizePolicy::Preferred</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="8,0,10">
                   <item>
                    <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0">
                     <item>
                      <widget class="QLabel" name="label_2">
                       <property name="text">
                        <string>组件树：</string>
                       </property>
                      </widget>
                     </item>
                     <item>
                      <widget class="QTreeView" name="treeView">
                       <property name="minimumSize">
                        <size>
                         <width>220</width>
                         <height>0</height>
                        </size>
                       </property>
                       <property name="maximumSize">
                        <size>
                         <width>310</width>
                         <height>16777215</height>
                        </size>
                       </property>
                       <property name="allColumnsShowFocus">
                        <bool>false</bool>
                       </property>
                       <property name="headerHidden">
                        <bool>false</bool>
                       </property>
                       <attribute name="headerVisible">
                        <bool>true</bool>
                       </attribute>
                       <attribute name="headerCascadingSectionResizes">
                        <bool>true</bool>
                       </attribute>
                      </widget>
                     </item>
                    </layout>
                   </item>
                   <item>
                    <widget class="QFrame" name="frame">
                     <property name="minimumSize">
                      <size>
                       <width>0</width>
                       <height>0</height>
                      </size>
                     </property>
                     <property name="frameShape">
                      <enum>QFrame::StyledPanel</enum>
                     </property>
                     <property name="frameShadow">
                      <enum>QFrame::Raised</enum>
                     </property>
                     <layout class="QVBoxLayout" name="verticalLayout_5">
                      <item>
                       <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,0,0,0,0,0,0">
                        <item>
                         <spacer name="verticalSpacer">
                          <property name="orientation">
                           <enum>Qt::Vertical</enum>
                          </property>
                          <property name="sizeType">
                           <enum>QSizePolicy::Preferred</enum>
                          </property>
                          <property name="sizeHint" stdset="0">
                           <size>
                            <width>20</width>
                            <height>40</height>
                           </size>
                          </property>
                         </spacer>
                        </item>
                        <item>
                         <widget class="QLineEdit" name="lineEdit">
                          <property name="maxLength">
                           <number>15</number>
                          </property>
                          <property name="placeholderText">
                           <string>系列名称</string>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <widget class="QPushButton" name="pushButton_5">
                          <property name="text">
                           <string>生成树</string>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <widget class="QPushButton" name="pushButton_2">
                          <property name="text">
                           <string>新建</string>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <widget class="QPushButton" name="pushButton">
                          <property name="text">
                           <string>删除</string>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <widget class="QPushButton" name="pushButton_4">
                          <property name="toolTip">
                           <string>Ctrl 标记父组件位置</string>
                          </property>
                          <property name="text">
                           <string>指定位置</string>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <widget class="QPushButton" name="pushButton_3">
                          <property name="minimumSize">
                           <size>
                            <width>0</width>
                            <height>35</height>
                           </size>
                          </property>
                          <property name="text">
                           <string>生成组件</string>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <spacer name="verticalSpacer_2">
                          <property name="orientation">
                           <enum>Qt::Vertical</enum>
                          </property>
                          <property name="sizeHint" stdset="0">
                           <size>
                            <width>20</width>
                            <height>40</height>
                           </size>
                          </property>
                         </spacer>
                        </item>
                       </layout>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLabel" name="label_3">
                     <property name="minimumSize">
                      <size>
                       <width>5</width>
                       <height>0</height>
                      </size>
                     </property>
                     <property name="text">
                      <string/>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer_7">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeType">
                    <enum>QSizePolicy::Expanding</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </widget>
               <widget class="QWidget" name="tab_5">
                <attribute name="title">
                 <string>可用角色</string>
                </attribute>
               </widget>
               <widget class="QWidget" name="tab_4">
                <attribute name="title">
                 <string>包含组件</string>
                </attribute>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="tab_2">
            <attribute name="title">
             <string>引擎配置</string>
            </attribute>
           </widget>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QProgressBar" name="progressBar">
             <property name="maximumSize">
              <size>
               <width>1000</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="value">
              <number>0</number>
             </property>
             <property name="textVisible">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_6">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QSlider" name="horizontalSlider">
             <property name="minimum">
              <number>50</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>100</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>tabWidget_2</tabstop>
  <tabstop>treeView</tabstop>
  <tabstop>lineEdit</tabstop>
  <tabstop>pushButton_5</tabstop>
  <tabstop>pushButton_2</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>pushButton_4</tabstop>
  <tabstop>pushButton_3</tabstop>
  <tabstop>horizontalSlider</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
